versions/schematic-2.md

```diff
@@ -0,0 +1,235 @@
+
+##### Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="schematicVersion"></a>Version | `NBT int` | **Required.** Specified the format version being used. It may be used to provide validation and auto-conversion from older versions. The current version is `2`. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicContentVersion"></a>ContentVersion | `NBT int` | **Required.** A version identifier for the contents. Representing the minecraft version, when serialized. Used for providing backwards compatibility. [Values are specified by Mojang](https://minecraft-de.gamepedia.com/Versionen#Version-IDs). Mojangs data converter for older versions will need this value for proper conversion. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicMetadata"></a>Metadata | [Metadata Object](#metadataObject) as `NBT compound` | **Optional.** Provides metadata about the schematic. This shall be saved as [NBT](#formFile) tag compound.
+<a name="schematicWidth"></a>Width | `NBT int` | **Required.** Specifies the width (the amount of blocks along the X-axis) of the schematic, must be interpreted as unsigned. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicHeight"></a>Height | `NBT int` | **Required.** Specifies the height (the amount of blocks along the Y-axis) of the schematic, must be interpreted as unsigned. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicLength"></a>Length | `NBT int` | **Required.** Specifies the length (the amount of blocks along the Z-axis) of the schematic, must be interpreted as unsigned. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicOffset"></a>Offset | `NBT int array`[3] | **Optional.** Specifies the relative offset of the schematic. This value is relative to the most minimal point in the schematics area. If present, this offset shall be added to all blocks, tiles and entities when pasting it into the world (may be negative). This shall be saved as [NBT](#formFile) tag int array.
+<a name="schematicPaletteMax"></a>PaletteMax | `NBT byte` | **Required.** Specifies the size of the block palette index in number of bytes needed for the maximum palette index. Implementations must use this to determine, which index format will be needed. The palette index may fit within a datatype smaller than a 32-bit integer or even a longer datatype like 64-bit integer. This shall be saved as [NBT](#formFile) tag byte.
+<a name="schematicPalette"></a>Palette | `NBT list` of [Palette Object](#paletteObject) as `NBT compound` | **Required.** Specifies the block palette. This is a mapping of block states to indices which are local to this schematic. The array element indices are used to reference the block states from within the [BlockData array](#schematicBlockData). The maximum index cannot be greater than [`2^(PaletteMax * 8) - 1`](#schematicPaletteMax).  This shall be saved as [NBT](#formFile) tag list of [NBT](#formFile) tag string.
```

Comment:
Yea, there i got a mistake. It's not a "NBT list of Palette Object as NBT compound", it's a "Palette Object as NBT list of NBT string". My bad.

I totally don't see why reading a list of string is harder than reading a key -> int id.

Basic problem here is, how would i get a block at x, y, z coordinate?
I don't know its id, so i can't just pick it out a map of key -> int id, because i don't know the key.
I would need to iterate over the whole map until i find the int id value to grab the key.

With the new system list of string, i can just calculate the index and get the element at this index of the list (or array).
Schematics always used this system to calculate list / array indices and just directly pick out what you need without searching.

I really hardly recommend to keep this huge advantage of legacy schematics, because i don't see anything bad at it.